akka-stream-tests/src/test/scala/akka/stream/scaladsl/FramingSpec.scala

```diff
@@ -174,6 +174,30 @@ class FramingSpec extends AkkaSpec {
         3.seconds) should ===(Vector.empty)
     }
 
+    "work with grouped frames" in {
+
+      for {
+        byteOrder ← byteOrders
+        fieldOffset ← fieldOffsets
+        fieldLength ← fieldLengths
+        count ← 2 to frameLengths.count(_ < (1L << (fieldLength * 8)))
+      } {
+
+        val encodedFrames = frameLengths.filter(_ < (1L << (fieldLength * 8))).map { length ⇒
+          val payload = referenceChunk.take(length)
+          encode(payload, fieldOffset, fieldLength, byteOrder)
+        }
+        val grouped = encodedFrames.grouped(count).map(_.fold(ByteString.empty)((r, bs) ⇒ r ++ bs)).toList
+
+        Await.result(
+          Source(encodedFrames)
+            .via(Framing.lengthField(fieldLength, fieldOffset, Int.MaxValue, byteOrder))
+            .grouped(10000)
+            .runWith(Sink.head),
+          3.seconds) should ===(encodedFrames)
+      }
+    }
+
```

Comment:
I add a simple unit test using `TestPublisher` and `TestSubscriber`. I did reproduce the failure. Thanks.